compiler/AST/AstPrintDocs.cpp

```diff
@@ -78,6 +78,12 @@ bool AstPrintDocs::enterFnSym(FnSymbol* node) {
 
 
 bool AstPrintDocs::enterModSym(ModuleSymbol* node) {
+  // If a module is not supposed to be documented, do not traverse into it (and
+  // skip the documentation).
+  if (node->hasFlag(FLAG_NO_DOC)) {
+      return false;
+  }
+
```

Comment:
Is it still useful for node->printDocs to do this same check?  I'd remove it from there, otherwise.